src/transformer.ts

```diff
@@ -1,21 +1,26 @@
 import type { Transformer } from "./deps.deno.ts";
 
-const wellKnownParseModesMap = new Map([
-  ["html", "HTML"],
-  ["markdown", "Markdown"],
-  ["markdownv2", "MarkdownV2"],
-]);
+const knownParseModes = {
+  "html": "HTML",
+  "markdown": "Markdown",
+  "markdownv2": "MarkdownV2",
+} as const;
```

Comment:
```suggestion
const knownParseModes = {
  "HTML": "html",
  "Markdown": "markdown",
  "MarkdownV2": "markdownv2",
} as const;
```

By doing this we avoid the need for `toLowerCase`, simplify the `parseMode`  parameter type to `keyof typeof knownParseModes` and avoid the need for overloading, which in turn avoids the need to change the `buildTransformer`  function into a normal function

Reply:
Heads up: this suggestion alone doesn't work. If you like it, don't forget to change the rest of the code too :)

Reply:
But then `knownParseModes` map becomes meaningless, since I thought it is used to normalized user's input "html" into "HTML".

Reply:
That will still be the case, because you'll. pass HTML to the plug-in and it'll send html to telegram

Reply:
Telegram converts everything to lowercase anyway. We enforce the right casing through types because it's desirable to write code that is consistent with the documentation.

Reply:
So, should I change something?

Reply:
In the end, it would be nice to see bot code that uses `HTML` and `MarkdownV2` and so on. That's the values we find in the Telegram docs, that's what's in the grammY types, and that's what most bots use today.

I think it's possible to get rid of the mapping entirely, and only do something like
```ts
type KnownParseModes = "HTML" | "Markdown" | "MarkdownV2"
function buildTransformer(parseMode: KnownParseModes) { }
```
What do you think?

Reply:
Fully agree, code looks much cleaner. But what if someone passes `"html"` in their code, won't this typing break things? Or it is a small issue?

Reply:
If they use TS, then auto-complete will change it to `HTML` anyway. If they choose to ignore this, then TS will report and error and tell them to use `HTML`, which is good style.

If they don't use TS, then they could end up writing stuff like `html` or `hTmL` or whatever, and it will still work.